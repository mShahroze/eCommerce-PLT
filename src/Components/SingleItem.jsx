import React from "react";
import "./Styles/SingleItem.css";
import { connect } from "react-redux";
import { addToBasket } from "../Redux/Shopping/shopping-actions";

const SingleItem = ({ currentItem, addToBasket }) => {
  return (
    <div className="singleItem">
      <img
        className="singleItemImage"
        src={currentItem.img}
        alt={currentItem.name}
      />
      <div className="singleItemDetails">
        <p className="detailsName">{currentItem.name}</p>
        <p className="detailsColour">{currentItem.colour}</p>
        <p className="detailsPrice">Â£ {currentItem.price}</p>

        <button
          data-testid="addToBasket"
          onClick={() => addToBasket(currentItem.id)}
          className="detailsAddBtn"
        >
          Add To Cart
        </button>
      </div>
    </div>
  );
};

const mapStateToProps = (state) => {
  return {
    currentItem: state.shop.currentItem,
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    addToBasket: (id) => dispatch(addToBasket(id)),
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(SingleItem);
