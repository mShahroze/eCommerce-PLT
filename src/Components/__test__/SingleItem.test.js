import React from "react";
import SingleItem from "../SingleItem";
import { fireEvent, render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initialState = {
  shop: {
    products: [],
    basket: [],
    currentItem: {
      id: 1,
      colour: "Black",
      name: "Black Sheet Strappy Textured Glitter Bodycon Dress",
      price: 10,
      img: "http://cdn-img.prettylittlething.com/9/0/a/a/90aa90903a135ee59594f47c7685aa7ef3046e44_cly8063_1.jpg?imwidth=1024",
      quantity: 1,
    },
    loading: true,
  },
};
const store = mockStore(initialState);

describe("SingleItem", () => {
  test("renders correctly", () => {
    const { container } = render(
      <Provider store={store}>
        <BrowserRouter>
          <SingleItem product={initialState.shop.products[0]} />
        </BrowserRouter>
      </Provider>
    );

    expect(container);
  });

  test("matches snapshot", () => {
    const { asFragment } = render(
      <Provider store={store}>
        <BrowserRouter>
          <SingleItem product={initialState.shop.products[0]} />
        </BrowserRouter>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test("adds Item to Basket", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <BrowserRouter>
          <SingleItem product={initialState.shop.products[0]} />
        </BrowserRouter>
      </Provider>
    );
    const addToBasketEL = getByTestId("addToBasket");
    fireEvent.click(addToBasketEL);
    const calledActions = store.getActions();
    expect(calledActions[0].type).toBe("ADD_TO_BASKET");
  });
});
