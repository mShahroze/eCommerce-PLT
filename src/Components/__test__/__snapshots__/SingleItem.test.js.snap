// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleItem matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="singleItem"
  >
    <img
      alt="Black Sheet Strappy Textured Glitter Bodycon Dress"
      class="singleItemImage"
      src="http://cdn-img.prettylittlething.com/9/0/a/a/90aa90903a135ee59594f47c7685aa7ef3046e44_cly8063_1.jpg?imwidth=1024"
    />
    <div
      class="singleItemDetails"
    >
      <p
        class="detailsName"
      >
        Black Sheet Strappy Textured Glitter Bodycon Dress
      </p>
      <p
        class="detailsColour"
      >
        Black
      </p>
      <p
        class="detailsPrice"
      >
        Â£ 10
      </p>
      <button
        class="detailsAddBtn"
        data-testid="addToBasket"
      >
        Add To Cart
      </button>
    </div>
  </div>
</DocumentFragment>
`;
